import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java

```diff
@@ -90,116 +84,80 @@ protected void store(
-          });
-        }
+      List<StreamProcessor> processorList = new ArrayList<>();
+      if (redactor != null) {
+        processorList.add(redactor::redact);
+      }
 
+      if(transcodeToSyntax != null && !transcodeToSyntax.equals(transferSyntax)) {
         processorList.add((inputStream, outputStream) -> {
-          destinationClient.get().stowRs(inputStream);
+          try (Transcoder transcoder = new Transcoder(inputStream)) {
+            transcoder.setIncludeFileMetaInformation(true);
+            transcoder.setDestinationTransferSyntax(transcodeToSyntax);
+            transcoder.transcode((transcoder1, dataset) -> outputStream);
+          }
         });
-
-        processStream(association.getApplicationEntity().getDevice().getExecutor(), inWithHeader, processorList);
-        uploadedBytesCount = countingStream.getCount();
-
-        updateResponseToSuccess(response, uploadedBytesCount);
-
-        if (backupUploadService != null && backupState.get() != null) {
-          backupUploadService.removeBackup(backupState.get().getUniqueFileName());
-        }
-      } catch (DicomWebException dwe) {
-        if (backupUploadService != null
-            && backupUploadService.filterHttpCode(dwe.getHttpStatus())
-            && backupState.get().getAttemptsCountdown() > 0) {
-          log.error("C-STORE request failed. Trying to resend...", dwe);
-          resendWithDelayRecursivelyExceptionally(backupState, destinationClient);
-          updateResponseToSuccess(response, uploadedBytesCount);
-          return;
-        }
-        reportError(dwe);
-        throw new DicomServiceException(dwe.getStatus(), dwe);
-      } catch (IBackupUploader.BackupException e) {
-        MonitoringService.addEvent(Event.CSTORE_BACKUP_ERROR);
-        log.error("Backup io processing during C-STORE request is failed: ", e);
-        reportError(e);
-        throw new DicomServiceException(Status.ProcessingFailure, e);
-      } catch (DicomServiceException e) {
-        reportError(e);
-        throw e;
-      } catch (Throwable e) {
-        reportError(e);
-        throw new DicomServiceException(Status.ProcessingFailure, e);
       }
-  }
 
-  private void updateResponseToSuccess(Attributes response, long countingStreamCount) {
-    response.setInt(Tag.Status, VR.US, Status.Success);
-    MonitoringService.addEvent(Event.CSTORE_BYTES, countingStreamCount);
-  }
+      if (multipleSendService != null) {
+        processorList.add((inputStream, outputStream) -> {
+          multipleSendService.start(
+              destinationHolder.getHealthcareDestinations(),
+              destinationHolder.getDicomDestinations(),
+              inputStream,
+              sopClassUID,
+              sopInstanceUID
+            );
+        });
+      } else {
+        processorList.add((inputStream, outputStream) -> {
+          destinationHolder.getSingleDestination().stowRs(inputStream);
+        });
```

Comment:
We need to still support the backup service in the case where we do not send to multiple destinations

Reply:
it`s already present in MultipleDestinationUploadService with SingleDestinationClientFactory. (configuration of this mode fixed in ImportAdapter now)